src/components/anti-afk.ts

```diff
@@ -0,0 +1,35 @@
+import { strict as assert } from "assert";
+
+import * as Discord from "discord.js";
+
+import { SECOND } from "../common.js";
+import { BotComponent } from "../bot-component.js";
+
+export default class AntiAFK extends BotComponent {
+    private countdown = new Map<string, NodeJS.Timeout>();
+
+    override async on_voice_state_update(old_state: Discord.VoiceState, new_state: Discord.VoiceState) {
+        if ((!new_state.selfDeaf || new_state.channel != old_state.channel) && this.countdown.has(new_state.id)) {
+            clearTimeout(this.countdown.get(new_state.id));
+            this.countdown.delete(new_state.id);
+        }
+        if (
+            new_state.channel != null &&
+            new_state.channelId != this.wheatley.guild.afkChannelId &&
+            new_state.selfDeaf
+        ) {
+            assert(new_state.member);
+            const member = new_state.member;
+            const timeout = setTimeout(
```

Comment:
Take a look at the set_timeout utility, in general the bot uses this and similar everywhere. It's mostly for an old debugging reason but I think the instrumentation is valuable, and consistency etc.

Reply:
fixed